src/main/kotlin/no/nav/familie/ef/sak/api/simulering/SimuleringService.kt

```diff
@@ -0,0 +1,70 @@
+package no.nav.familie.ef.sak.api.simulering
+
+import no.nav.familie.ef.sak.api.beregning.VedtakService
+import no.nav.familie.ef.sak.iverksett.IverksettClient
+import no.nav.familie.ef.sak.repository.domain.Behandling
+import no.nav.familie.ef.sak.repository.domain.BehandlingType
+import no.nav.familie.ef.sak.repository.domain.Fagsak
+import no.nav.familie.ef.sak.service.BehandlingService
+import no.nav.familie.ef.sak.service.FagsakService
+import no.nav.familie.ef.sak.service.TilkjentYtelseService
+import no.nav.familie.ef.sak.sikkerhet.SikkerhetContext
+import no.nav.familie.kontrakter.ef.iverksett.SimuleringDto
+import no.nav.familie.kontrakter.felles.simulering.DetaljertSimuleringResultat
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+import java.util.UUID
+
+@Service
+class SimuleringService(private val iverksettClient: IverksettClient,
+                        private val behandlingService: BehandlingService,
+                        private val fagsakService: FagsakService,
+                        private val vedtakService: VedtakService,
+                        private val blankettSimuleringsService: BlankettSimuleringsService,
+                        private val tilkjentYtelseService: TilkjentYtelseService) {
+
+
+    fun simulerForBehandling(behandlingId: UUID): SimuleringsresultatDto {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val fagsak = fagsakService.hentFagsak(behandling.fagsakId)
+
+        val simuleringResultat = when (behandling.type) {
+            BehandlingType.BLANKETT -> simulerUtenTilkjentYtelse(behandling, fagsak)
+            else -> simulerMedTilkjentYtelse(behandling, fagsak)
+        }
+
+        val datoForSimulering = LocalDate.now() // TODO: bruk databaseverdi hvis resultatet er hentet fra DB
```

Comment:
Steg 2 for denne oppgaven er å lagre simuleringsverdier i databasen slik at vi kan se på simuleringsverdien i historisk perspektiv. Da må denne datoen tas vare på og brukes i mappingen. Siden vi nå henter simulering hver gang funker det greit med `now`

Reply:
En per behandling historiskt eller flere historiske per behandling?

Reply:
èn per behandling er nok "korrekt" i denne sammenhengen, tror jeg